applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -20,148 +20,84 @@
-void ComputeHessianSolMetricProcess<TDim, TVarType>::Execute()
+void ComputeHessianSolMetricProcess::Execute()
 {
-    // Iterate in the nodes
-    NodesArrayType& nodes_array = mThisModelPart.Nodes();
-    const int num_nodes = nodes_array.end() - nodes_array.begin();
-
     CalculateAuxiliarHessian();
 
     // Some checks
-    VariableUtils().CheckVariableExists(mVariable, nodes_array);
+    NodesArrayType& nodes_array = mThisModelPart.Nodes();
+    if (mrOriginVariableDoubleList.size() > 0) {
+        VariableUtils().CheckVariableExists(mrOriginVariableDoubleList[0], nodes_array);
+    } else {
+        VariableUtils().CheckVariableExists(mrOriginVariableComponentsList[0], nodes_array);
+    }
     for (auto& i_node : nodes_array)
         KRATOS_ERROR_IF_NOT(i_node.Has(NODAL_H)) << "NODAL_H must be computed" << std::endl;
 
-    // Ratio reference variable
-    KRATOS_ERROR_IF_NOT(KratosComponents<Variable<double>>::Has(mRatioReferenceVariable)) << "Variable " << mRatioReferenceVariable << " is not a double variable" << std::endl;
-    const auto& reference_var = KratosComponents<Variable<double>>::Get(mRatioReferenceVariable);
-
-    // Tensor variable definition
-    const Variable<TensorArrayType>& tensor_variable = KratosComponents<Variable<TensorArrayType>>::Get("METRIC_TENSOR_"+std::to_string(TDim)+"D");
-
-    // Setting metric in case not defined
-    if (!nodes_array.begin()->Has(tensor_variable)) {
-        // Declaring auxiliar vector
-        const TensorArrayType aux_zero_vector = ZeroVector(3 * (TDim - 1));
-        #pragma omp parallel for
-        for(int i = 0; i < num_nodes; ++i) {
-            auto it_node = nodes_array.begin() + i;
-            it_node->SetValue(tensor_variable, aux_zero_vector);
-        }
-    }
-
-    #pragma omp parallel for
-    for(int i = 0; i < num_nodes; ++i) {
-        auto it_node = nodes_array.begin() + i;
-
-        const Vector& hessian = it_node->GetValue(AUXILIAR_HESSIAN);
-
-        const double nodal_h = it_node->GetValue(NODAL_H);
-
-        double element_min_size = mMinSize;
-        if ((element_min_size > nodal_h) && mEnforceCurrent) element_min_size = nodal_h;
-        double element_max_size = mMaxSize;
-        if ((element_max_size > nodal_h) && mEnforceCurrent) element_max_size = nodal_h;
-
-        // Isotropic by default
-        double ratio = 1.0;
-
-        if (it_node->SolutionStepsDataHas(reference_var)) {
-            const double ratio_reference = it_node->FastGetSolutionStepValue(reference_var);
-            ratio = CalculateAnisotropicRatio(ratio_reference, mAnisotropicRatio, mBoundLayer, mInterpolation);
-        }
-
-        // For postprocess pourposes
-        it_node->SetValue(ANISOTROPIC_RATIO, ratio);
+    const auto& it_element_begin = mThisModelPart.ElementsBegin();
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t dimension = r_first_element_geometry.WorkingSpaceDimension();
```

Comment:
I would say the processinfo is safer

Reply:
For what?

Reply:
determining the dimension of the problem